observable.go

```diff
@@ -1601,6 +1602,29 @@ func (o *observable) TakeWhile(apply Predicate) Observable {
 	return newColdObservableFromFunction(f)
 }
 
+func (o *observable) TimeInterval() Observable {
+	f := func(out chan interface{}) {
+		it := o.Iterator(context.Background())
+		var latestTime time.Time
+		for {
+			if _, err := it.Next(context.Background()); err == nil {
+				if latestTime.IsZero() {
+					out <- time.Duration(0)
```

Comment:
I'm not sure about sending this very item. According to the [documentation](http://reactivex.io/documentation/operators/timeinterval.html), the first item sent should rather be the duration between the subscription and the first item received.
In this regards, we should rather initialize `latestTime` to `time.Now()` in my opinion and get rid of this if/else statement.
What do you think? 

Reply:
This should impact the tests as well.

Reply:
Nice, thanks for the great feedback @teivah. I wasn't sure how to represent the time of the subscription, but it's clearer now. Will fix as soon as I can.